### PR TITLE
[processing] Remove hardcoded v7 from grass provider, some fixes for v8

### PR DESCRIPTION
**Pull request by nyalldawson**
The provider works with grass versions > 7, so hardcoding the older version name in is not appropriate. 

Issue raised in GRASS GIS meeting, June 2023

### CODE REVIEW

File: src/core/processing/qgsprocessingregistry.cpp [hunk 1]
```diff
@@ -156,9 +156,17 @@ bool QgsProcessingRegistry::removeProvider( const QString &providerId )
...
-QgsProcessingProvider *QgsProcessingRegistry::providerById( const QString &id )
+QgsProcessingProvider *QgsProcessingRegistry::providerById( const QString &id ) const
 {
-  return mProviders.value( id, nullptr );
+  auto it = mProviders.constFind( id );
+  if ( it != mProviders.constEnd() )
+    return it.value();
+
+  // transparently map old references to "grass7" provider to "grass" provider
+  if ( id.compare( QLatin1String( "grass7" ), Qt::CaseInsensitive ) == 0 )
```
Comment (lbartoletti) [inline src/core/processing/qgsprocessingregistry.cpp:166]: This condition is required to run old scripts, right?

File: python/plugins/grassprovider/tests/testdata/grass_algorithms_imagery_tests.yaml [hunk 2]
```diff
@@ -328,7 +328,9 @@ tests:
         type: raster
     results:
       imaginary:
-        hash: 94249384dd8b6019f0024501bc9a093cba9dd025c183d3fb46d77027
+        hash:
```
Comment (lbartoletti) [inline python/plugins/grassprovider/tests/testdata/grass_algorithms_imagery_tests.yaml:331]: why changes here?

**Review by lbartoletti — APPROVED** [review]
lgtm, only two questions.

**Comment by lbartoletti** [comment]
Thanks, for taking this action!

File: python/plugins/grassprovider/tests/testdata/grass_algorithms_imagery_tests.yaml (hunk 2 above)
Comment (nyalldawson) [reply python/plugins/grassprovider/tests/testdata/grass_algorithms_imagery_tests.yaml:331]: Newer grass is giving slightly different results for the algorithms, so it's to get the tests passing on newer versions

Comment (nyalldawson) [reply python/plugins/grassprovider/tests/testdata/grass_algorithms_imagery_tests.yaml:331]: (the hash list is a list of acceptable values)

File: src/core/processing/qgsprocessingregistry.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/processing/qgsprocessingregistry.cpp:166]: Yep, and old models

File: python/plugins/grassprovider/tests/testdata/grass_algorithms_imagery_tests.yaml (hunk 2 above)
Comment (lbartoletti) [reply python/plugins/grassprovider/tests/testdata/grass_algorithms_imagery_tests.yaml:331]: Ok, thanks. That's what I thought, but I wanted confirmation.

**Comment by florisvdh** [comment]
Cool! Confirmed with `qgis_process` and the R package.

**Pull request merged**